week4-task/src/pages/Signup/Signup.tsx

```diff
@@ -0,0 +1,120 @@
+import { useState } from "react";
+import { useNavigate, Link } from "react-router-dom";
+import { useForm } from "@hooks/useForm";
+import { signup } from "@api/auth";
+import SignupIdStep from "./components/SignupIdStep";
+import SignupPasswordStep from "./components/SignupPasswordStep";
+import SignupNicknameStep from "./components/SignupNicknameStep";
+import {
+  container,
+  title,
+  linkText,
+  bottomSection,
+} from "@shared/styles/formCommon.css";
+import {
+  isIdValid,
+  isPasswordValid,
+  isNicknameValid,
+  getIdError,
+  getPasswordError,
+} from "@utils/validation";
+
+const Signup = () => {
+  const navigate = useNavigate();
+  const [step, setStep] = useState(1);
+  const [form, handleInputChange] = useForm({
+    id: "",
+    password: "",
+    passwordCheck: "",
+    nickname: "",
+  });
+  const [showPassword, setShowPassword] = useState(false);
+
+  const handleToggleShowPassword = () => {
+    setShowPassword((prev) => !prev);
+  };
+
+  // 유효성 검사, 에러 메시지
+  const idError = getIdError(form.id);
+  const passwordError = getPasswordError(form.password, form.passwordCheck);
+
+  const handleNextClick = () => {
+    if (step === 1 && isIdValid(form.id)) {
+      setStep(2);
+    } else if (
+      step === 2 &&
+      isPasswordValid(form.password, form.passwordCheck)
+    ) {
+      setStep(3);
+    }
+  };
+
+  const handleSignup = async () => {
+    if (isNicknameValid(form.nickname)) {
+      const res = await signup({
+        loginId: form.id,
+        password: form.password,
+        nickname: form.nickname,
+      });
+      if (res.success) {
+        alert(`${res.data?.nickname} 님 회원가입 성공하셨습니다!`);
+        navigate("/login");
+      } else {
+        alert(res.message);
+      }
+    }
+  };

```

Comment:
```suggestion
const handleSignup = async () => {
  if (!isNicknameValid(form.nickname)) return;

  const res = await signup({
    loginId: form.id,
    password: form.password,
    nickname: form.nickname,
  });

  if (res.success) {
    alert(`${res.data?.nickname} 님 회원가입 성공하셨습니다!`);
    navigate("/login");
    return;
  }

  alert(res.message);
};
```
중첩된 if문이 가독성을 해치고 있어서 early return으로 개선해보는건 어떨까요